crates/runtime/src/init/dataset.rs

```diff
@@ -649,8 +666,20 @@ impl Runtime {
         // Remove datasets that are no longer in the app
         for ds in &current_app.datasets {
             if !new_app.datasets.iter().any(|d| d.name == ds.name) {
-                let ds = match Dataset::try_from(ds.clone()) {
-                    Ok(ds) => ds,
+                let ds = match DatasetBuilder::try_from(ds.clone()) {
```

Comment:
It feels a little weird that we need to construct a new dataset object to be able to remove it from being registered. Could we update `self.remove_dataset` to instead just take the dataset name?